src/Domain/DonationRepository.php

```diff
@@ -179,52 +178,36 @@ public function buildFromApiRequest(DonationCreate $donationData): Donation
-        $allocationTries = 0;
         // We look up matching withdrawals to allow for the case where retrospective matching was required
         // and the donation is not new, and *some* (or full) matching already occurred. The collection of withdrawals
         // is most often empty (for new donations) so this will frequently be 0.00.
         $amountMatchedAtStart = $donation->getFundingWithdrawalTotal();
 
-        /** @var FundingWithdrawal[] $newWithdrawals */
-        $newWithdrawals = [];
-
-        while (!$allocationDone && $allocationTries < $this->maxLockTries) {
-            try {
-                $likelyAvailableFunds = $this->getEntityManager()
-                    ->getRepository(CampaignFunding::class)
-                    ->getAvailableFundings($donation->getCampaign());
+        try {
+            $likelyAvailableFunds = $this->getEntityManager()
+                ->getRepository(CampaignFunding::class)
+                ->getAvailableFundings($donation->getCampaign());
 
-                foreach ($likelyAvailableFunds as $funding) {
-                    if ($funding->getCurrencyCode() !== $donation->getCurrencyCode()) {
-                        throw new \UnexpectedValueException('Currency mismatch');
-                    }
+            foreach ($likelyAvailableFunds as $funding) {
+                if ($funding->getCurrencyCode() !== $donation->getCurrencyCode()) {
+                    throw new \UnexpectedValueException('Currency mismatch');
                 }
-
-                $lockStartTime = microtime(true);
-                $newWithdrawals = $this->matchingAdapter->runTransactionally(
-                    fn() => $this->safelyAllocateFunds($donation, $likelyAvailableFunds, $amountMatchedAtStart)
-                );
-                $lockEndTime = microtime(true);
-
-                $allocationDone = true;
-                $this->persistQueuedDonations();
-            } catch (Matching\TerminalLockException $exception) { // Includes non-retryable `DBALException`s
-                $waitTime = round(microtime(true) - $lockStartTime, 6);
-                $this->logError(
-                    "Match allocate FINAL error: ID {$donation->getUuid()} got " . get_class($exception) .
-                    " after {$waitTime}s on try #$allocationTries: {$exception->getMessage()}"
-                );
-                throw $exception; // Re-throw exception after logging the details if not recoverable
             }
-        }
 
-        if (!$allocationDone) {
-            $this->logger->error(
-                "Match allocate FINAL error: ID {$donation->getUuid()} failed matching after $allocationTries tries"
+            $lockStartTime = microtime(true);
+            $newWithdrawals = $this->matchingAdapter->runTransactionally(
+                fn() => $this->safelyAllocateFunds($donation, $likelyAvailableFunds, $amountMatchedAtStart)
             );
-
-            throw new DomainLockContentionException();
```

Comment:
I think we've removed all places we threw `DomainLockContentionException` now. So we can also remove the exception itself and the 1 place we catch it, in `MatchBot\Application\Actions\Donations\Create`.

Reply:
Yeah will look at deleting that on Monday. I wonder if Psalm should have an issue for an unused class. You might have to configure or mark any classes that have usages that Psalm can't detect or that are intended for future use.